### PR TITLE
Adds in Our Galaxy Realization Ability

### PR DESCRIPTION
**Pull request by TheBronJameOffical**
## About The Pull Request

Adds in an ability for Our Galaxy, the CotG Realization.
On activation, gives any (alive) allies in a 6 tile radius a temporary and more powerful version of CotG's pebble.
This version will last for 30 seconds and scales 0.5 healing per person with the gift (Minimum of 1.5).
As a drawback, the User of the ability is more important as their death will cause a similar effect when enough of CotG's friends die. Dealing Black damage to all with the gift based on how many people have it.
Comes with a 1 minute cooldown and does not apply to people who are working.

Also tweaks Our Galaxy's armor values to 4/8/8/6.

## Why It's Good For The Game

Finishes what I started, now you too can be the healer.

## Changelog
:cl:
add: Added Our Galaxy realization ability
tweak: tweaked Our Galaxy armor values
imageadd: added relevant iconsimageadd: added relevant icons
/:cl:

### CODE REVIEW

**Comment by CrabbytheCrab** [comment]
so like sakura bloom's ability but makes everyone gets punished instead of the user

File: code/modules/spells/ability_types/realized.dm [hunk 1]
```diff
@@ -862,3 +862,94 @@
...
+			if(L.status_flags & GODMODE)
+				continue
+			if(L.stat == DEAD)
+				continue
+			if(L.is_working)
+				continue
+			if(L == H) //excludes the giver
+				continue
+			L.apply_status_effect(/datum/status_effect/galaxy_gift)
+	return ..()
```
Comment (LanceSmites328) [inline code/modules/spells/ability_types/realized.dm:893]:
```suggestion
/obj/effect/proc_holder/ability/galaxy_gift/Perform(target, mob/living/carbon/human/user)
	if(!istype(user))
		return
	var/list/existing_gifted = list()
	for(var/mob/living/carbon/human/H in GLOB.player_list)
		if(H.has_status_effect(/datum/status_effect/galaxy_gift))
			existing_gifted += H
	playsound(get_turf(user), 'sound/abnormalities/despairknight/gift.ogg', 50, 0, 2) //placeholder, uses KoD blessing noise at the moment
	for(var/turf/T in view(range, user))
		new /obj/effect/temp_visual/galaxy_aura(T)
		for(var/mob/living/carbon/human/H in T)
			if(!user.faction_check_mob(H, FALSE))
				continue
			if(H.stat == DEAD)
				continue
			if(H.is_working)
				continue
			var/datum/status_effect/galaxy_gift/new_gift = H.apply_status_effect(/datum/status_effect/galaxy_gift)
			if(H == user)
				new_gift.watch_death = TRUE
			existing_gifted |= H
	for(var/mob/living/carbon/human/H in existing_gifted)
		var/datum/status_effect/galaxy_gift/gift = H.has_status_effect(/datum/status_effect/galaxy_gift)
		if(!gift)
			continue
		gift.gifted = existing_gifted
	return ..()
```

File: code/modules/spells/ability_types/realized.dm [hunk 2]
```diff
@@ -862,3 +862,94 @@
...
+	icon_state = "friendship"
+
+/datum/status_effect/galaxy_gift/tick()
+	var/mob/living/carbon/human/Y = owner
+	gifted = 1
+	for(var/mob/living/carbon/human/L in GLOB.player_list) //The Heal Code, thanks to Chiemi for making it
+		if(L.has_status_effect(/datum/status_effect/galaxy_gift))
+			gifted += 1
+	Y.adjustBruteLoss(-(base_heal_amt*gifted))
+	Y.adjustSanityLoss(-(base_heal_amt*gifted))
```
Comment (LanceSmites328) [inline code/modules/spells/ability_types/realized.dm:917]:
```suggestion
/datum/status_effect/galaxy_gift //the normal people gift
	id = "galaxygift"
	status_type = STATUS_EFFECT_UNIQUE
	duration = 30 SECONDS
	tick_interval = 1 SECONDS
	alert_type = /atom/movable/screen/alert/status_effect/galaxy_gift
	var/base_heal_amt = 0.5
	var/base_dmg_amt = 45
	var/watch_death = FALSE
	var/list/gifted

/atom/movable/screen/alert/status_effect/galaxy_gift
	name = "Parting Gift"
	desc = "You recover SP and HP over time temporarliy."
	icon = 'ModularTegustation/Teguicons/status_sprites.dmi'
	icon_state = "friendship"

/datum/status_effect/galaxy_gift/tick()
	if(!ishuman(owner))
		qdel(src)
		return
	var/mob/living/carbon/human/Y = owner
	listclearnulls(gifted)
	for(var/mob/living/carbon/human/H in gifted) //The Heal Code, thanks to Chiemi for making it
		if(H == Y)
			continue
		if(H.stat == DEAD || QDELETED(H))
			gifted -= H
			if(H) // If there's even anything left to remove
				H.remove_status_effect(/datum/status_effect/galaxy_gift)
	if(Y.stat == DEAD || QDELETED(Y))
		return watch_death ? Pop() : FALSE
	var/heal_mult = LAZYLEN(gifted)
	Y.adjustBruteLoss(-(base_heal_amt*heal_mult))
	Y.adjustSanityLoss(-(base_heal_amt*heal_mult))
	
/datum/status_effect/galaxy_gift/proc/Pop()
	var/damage_mult = LAZYLEN(gifted)
	for(var/mob/living/carbon/human/H in gifted)
		H.apply_damage(base_dmg_amt*damage_mult, BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
		H.remove_status_effect(/datum/status_effect/galaxy_gift)
		new /obj/effect/temp_visual/pebblecrack(get_turf(H))
		playsound(get_turf(H), "shatter", 50, TRUE)
		to_chat(H, "<span class='userdanger'>Your pebble violently shatters!</span>")
	return
```

File: code/modules/spells/ability_types/realized.dm [hunk 3]
```diff
@@ -862,3 +862,94 @@
...
+			if(L.has_status_effect(/datum/status_effect/galaxy_gift))
+				gifted += 1
+		for(var/mob/living/carbon/human/L in GLOB.player_list)
+			if(L.has_status_effect(/datum/status_effect/galaxy_gift))
+				L.apply_damage((base_dmg_amt*gifted), BLACK_DAMAGE, null, L.run_armor_check(null, BLACK_DAMAGE), spread_damage = TRUE)
+				L.remove_status_effect(/datum/status_effect/galaxy_gift)
+				new /obj/effect/temp_visual/pebblecrack(get_turf(L))
+				playsound(get_turf(L), "shatter", 50, TRUE)
+				to_chat(L, "<span class='userdanger'>Your pebble violently shatters!</span>")
+		Y.remove_status_effect(/datum/status_effect/galaxy_gift_origin)
```
Comment (LanceSmites328) [inline code/modules/spells/ability_types/realized.dm:955]:
```suggestion
```

**Review by LanceSmites328 — CHANGES_REQUESTED** [review]
This aint even a balance change review, we'll get to that in a sec. Don't make a status effect meant to be given out to groups of people at a time that loops through the entire player list a MINIMUM of once per second per instance. There's no need for that.

**Review by LanceSmites328 — APPROVED** [review]

**Pull request merged**